docker/dev/nextcloud/docker-compose.yml

```diff
@@ -0,0 +1,23 @@
+services:
+  nextcloud:
+    image: nextcloud:stable
+    restart: unless-stopped
+    networks:
+      - gateway
+    volumes:
+      - nextcloud:/var/www/html
+      #   If you want to use a local version of a plugin, mount a local folder
+      #   more detailed instructions available at https://github.com/nextcloud/integration_openproject/blob/master/docs/setup_via_docker.md
+      # - ../nextcloud_apps:/var/www/html/custom_apps
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.nextcloud.rule=Host(`nextcloud.local`)"
+      - "traefik.http.routers.nextcloud.entrypoints=websecure"
```

Comment:
There's also an option to using TLS here using

```
      - "traefik.http.routers.nextcloud.tls=true"
      - "traefik.http.routers.nextcloud.tls.certresolver=step"
```

But then we need to include and env variable

```
    environment:
      - OVERWRITEPROTOCOL=https
```

Reply:
Which purpose do these labels serve? Coming from the TLS configuration, I'd expect that using the `entrypoints=websecure` already doesn everything we need. For sure my nextcloud is served through HTTPS and step is providing the certificate for that with the config given in this PR.

As far as I can tell the settings you mention are set for the entrypoint already: https://github.com/opf/openproject/blob/dev/docker/dev/tls/docker-compose.yml#L14-L15

>  environment:
>    - OVERWRITEPROTOCOL=https

This is not necessary when the step about trusting Traefik has been done, because then NCs Apache will trust the `X-Forwarded-Proto` header provided by Traefik, which indicates the currently used protocol.

However, if you find a way to overwrite the trusted proxy with an env var, I'd be more than happy.

Reply:
I stopped tweaking the file when I got it working so I did not get to fine details. But fair point about the TLS. I assumed the wrong thing.

About the OVERWRITEPROTOCOL, I might have missed a previous step then. Cause I could only actually "upload" after adding it, otherwise the requests would always go to openproject.local by http. In any case it's all good.